### PR TITLE
Add GitHub action to update spackages :robot:

### PR DESCRIPTION
**Pull request by rbberger**
## PR Summary

This adds a manual and scheduled action to check for changes in ports-of-call and spiner spackages and migrate them over to singularity-eos. If a change is detected, it will create a PR authored by a bot.

## PR Checklist

- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Format your changes by using the `make format` command after configuring with `cmake`.
- [ ] Document any new features, update documentation for changes made.
- [ ] Make sure the copyright notice on any files you modified is up to date.
- [ ] After creating a pull request, note it in the CHANGELOG.md file
- [ ] If preparing for a new release, update the version in cmake.

### CODE REVIEW

File: .github/workflows/deps.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        - name: copy over spack-repo/packages
+          run: |
+            # assume ports-of-call may be newer than spiner
+            cp -R build/spiner/spack-repo/packages/* spack-repo/packages/
+            cp -R build/ports-of-call/spack-repo/packages/* spack-repo/packages/
+        - name: is there a difference?
+          run: git diff --exit-code --compact-summary
+          id: no_change
+        - name: create branch
+          if: ${{ failure() && steps.no_change.conclusion == 'failure' }}
```
Comment (jhp-lanl) [inline .github/workflows/deps.yml:42]: Forgive my ignorance, but what is the purpose of `failure()` here?

File: .github/workflows/deps.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          run: |
+            git config user.name 'github-actions[bot]'
+            git config user.email 'github-actions[bot]@users.noreply.github.com'
+            git branch -D update_spackages || true
+            git checkout -b update_spackages
```
Comment (jhp-lanl) [inline .github/workflows/deps.yml:47]: Should this branch have a more unique name to avoid name conflicts?

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -19,7 +19,7 @@ jobs:
 
       steps:      
         - name: Checkout code
-          uses: actions/checkout@v2
+          uses: actions/checkout@v3
```
Comment (jhp-lanl) [inline .github/workflows/docs.yml:22]: Forgive my ignorance, but why is this change necessary?

**Review by jhp-lanl — COMMENTED** [review]
I'm not really in a position to approve this MR but I did have some questions mostly for my own edification.

**Review by Yurlungur — APPROVED** [review]
Thanks for adding this @rbberger !

**Comment by Yurlungur** [comment]
I... *think* we probably don't need a changelog for this. Just ocurred to me that it might be good to add a sentence about the bot to the `contributing` section of the docs though.

File: .github/workflows/docs.yml (hunk 3 above)
Comment (rbberger) [reply .github/workflows/docs.yml:22]: Right now, with the old version you're getting a warning about an outdated node.js version. This removes it.

File: .github/workflows/deps.yml (hunk 2 above)
Comment (rbberger) [reply .github/workflows/deps.yml]: done.

File: .github/workflows/deps.yml (hunk 1 above)
Comment (rbberger) [reply .github/workflows/deps.yml]: just a copy&paste thing. It shouldn't be necessary, so I removed it.

**Review by jhp-lanl — APPROVED** [review]

**Comment by jhp-lanl** [comment]
@rbberger I'll merge after a successful pipeline unless there's more you want to add

**Comment by rbberger** [comment]
@Yurlungur @jhp-lanl added the note in the `contributing.rst`. That should be it.

**Comment by Yurlungur** [comment]
:+1: I think we can merge when tests pass then.
